codeflash/discovery/functions_to_optimize.py

```diff
@@ -470,6 +476,9 @@ def was_function_previously_optimized(
         Tuple of (filtered_functions_dict, remaining_count)
 
     """
+    if is_LSP_enabled():
```

Comment:
Q: could this raise questions on underterministic nature of optimizations if the function has is same.